sycl/test-e2e/AddressSanitizer/memory-leak/memory-leak-shared-lib.cpp

```diff
@@ -0,0 +1,37 @@
+// REQUIRES: linux, cpu
+// RUN: %{build} %device_asan_flags -O0 -g -fPIC -shared -DSHARED_LIB -o %t.so
+// RUN: %{build} %device_asan_flags -O0 -g -fPIC -Wl,-rpath,. %t.so -o %t
```

Comment:
Nit: No need for `-Wl,-rpath,.`, %t will always expand to a full path.

Reply:
We used -rpath here then we don't need to set env LD_LIBRARY_PATH.